src/caduceus/caduceus_test.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"errors"
+	"github.com/Comcast/webpa-common/health"
+	"github.com/Comcast/webpa-common/logging"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+// Begin mock declarations
+
+// mockHandler only needs to mock the `HandleRequest` method
+type mockHandler struct {
+	mock.Mock
+}
+
+func (m *mockHandler) HandleRequest(workerID int, inRequest CaduceusRequest) {
+	m.Called(workerID, inRequest)
+	return
+}
+
+// mockHealthTracker needs to mock things from both the `HealthTracker`
+// interface as well as the `health.Monitor` interface
+type mockHealthTracker struct {
+	mock.Mock
+}
+
+func (m *mockHealthTracker) SendEvent(healthFunc health.HealthFunc) {
+	m.Called(healthFunc)
+	return
+}
+
+func (m *mockHealthTracker) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	m.Called(response, request)
+	return
+}
+
+func (m *mockHealthTracker) IncrementBucket(inSize int) {
+	m.Called(inSize)
+	return
+}
+
+// Begin test functions
+
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func TestWorkerPool(t *testing.T) {
+	assert := assert.New(t)
+
+	workerPool := WorkerPoolFactory{
+		NumWorkers: 1,
+		QueueSize:  1,
+	}.New()
+
+	t.Run("TestWorkerPoolCreation", func(t *testing.T) {
+		assert.NotNil(workerPool)
+	})
+
+	t.Run("TestWorkerPoolSend", func(t *testing.T) {
+		err := workerPool.Send(func(workerID int) {
+			// do nothing
+		})
+
+		assert.Nil(err)
+	})
+
+	workerPool = WorkerPoolFactory{
+		NumWorkers: 1,
+		QueueSize:  0,
+	}.New()
+
+	t.Run("TestWorkerPoolFullQueue", func(t *testing.T) {
+		err := workerPool.Send(func(workerID int) {
+			// do nothing
+		})
+
+		assert.NotNil(err)
+	})
+}
+
+func TestIncrementBucket(t *testing.T) {
+	assert := assert.New(t)
+
+	testData := []struct {
+		inSize       int
+		expectedStat health.Stat
+	}{
+		{inSize: 99, expectedStat: PayloadsOverZero},
```

Comment:
You really want more values here.  It's a good idea to check a negative value, zero, and extremes like math.MaxInt32 or math.MinInt32.  The goal of that is to make sure that future changes to the code being tested cause test breaks where they should.  For example, what if someone adds a PayloadsOverFifty stat?

You can't be exhaustive, of course.  But spot checking more values makes the test better able to warn a maintenance programmer that he's done something that breaks your assumptions.